### ISSUE TITLE
add «aggressive» mode

### ISSUE THREAD

**Issue by Levitanus** [RomeoDespres/reapy#62]
I'm not sure reapy needs it, but, several times, yet I was wondering of wrapping `reapy.core` classes into «aggressive» decorators, which will invoke `make_current_project` and `connect` on initialization and attribute access. As well as `@inside_reaper` does.

I'm still not sure it is good idea, even if turned on within environment variable or config value, as sometimes we cannot control its influence on GUI, performance, while cannot guarantee stability in cases, when attributes are used after resolving. But, deeply in the soul I wish to write:
```Python
o_track = ss.MasterOutTrack(
    track=rpr.Track(id='out', project=rpr.Project(MASTER_PROJ_NAME)),
    slave=ss.SlaveProject(),
    target=ss.SlaveInTrack(
        rpr.Track(id='in', project=rpr.Project(SLAVE_PROJECT_NAME))
    )
)
```

instead of

```Python
m_pr = rpr.Project(MASTER_PROJ_NAME)
s_pr = rpr.Project(SLAVE_PROJECT_NAME)
with m_pr.make_current_project():
    pr = rpr.Project().name
    sleep(1)
    m_tr = rpr.Track(id='out', project=m_pr)
with s_pr.make_current_project():
    s_tr = rpr.Track(id='in', project=s_pr)
o_track = ss.MasterOutTrack(
    track=m_tr, slave=ss.SlaveProject(), target=ss.SlaveInTrack(s_tr)
)
```

**Comment by RomeoDespres** [comment]
I'm sorry but I don't understand your code... even the second part does not work in reapy.

First, you can't (yet) instantiate a Project by its name. You have to use the `index` argument. Assuming you have two open projects like below:

![image](https://user-images.githubusercontent.com/39157756/76980038-fd7af400-6938-11ea-9cd4-b16b7bc691d9.png)

you can access both projects as such:

```python
>>> master_project = reapy.Project(index=0)
>>> slave_project = reapy.Project(index=1)
>>> master_project.name
'Master.rpp'
>>> slave_project.name
'Slave.rpp'
```
Assuming that Master (resp. Slave) has a track named "out" (resp. in), you can access them as follows:

```python
>>> out_track = master_project.tracks['out']
>>> in_track = slave_project.tracks['in']
```

I don't know exactly what `ss` and `MasterOutTrack` are, but I don't think you need to set any project as "current" to do what you want to do.

**Comment by RomeoDespres** [comment]
Ups... Didn't see your PR about getting projects by their name. Still, I don't understand why you need to change the current project in your example.

**Comment by Levitanus** [comment]
> First, you can't (yet) instantiate a Project by its name.

Can at the #63 :)

> I'm sorry but I don't understand your code

I'm sorry, but the point is in using a couple of `reapy.Track` objects that belongs to the different projects. It's really a problem to access the track in the background project, as (I don't know actually, Reaper or reapy) does not response to it. So, only hook I've learned is to make everything related to project within making it `current`.

If I want to access some data of the `reapy.Track(id='(MediaTrack*)0xNNN...')` or `reapy.Track('name', project=<not the current project>)` I have to make the project be `current`. Didn't resolve is it Reaper or reapy behavior, but it exists.

**Comment by Levitanus** [comment]
Just tried within two project tabs and `test_master.RPP` activated
```Python
import reapy as rpr

master_project = rpr.Project('test_master')
master_project_track = master_project.tracks['out']
slave_project = rpr.Project('test_slave')
slave_project_track = slave_project.tracks['in']
# AttributeError: 'NoneType' object has no attribute 'id'

print(master_project_track.name)
print(slave_project_track.name)
```

**Comment by RomeoDespres** [comment]
Hmmm I see... This is a real bug that needs to be investigated. Thanks for spotting it.

**Comment by RomeoDespres** [comment]
Looks like it comes from the ReaScript API. Not sure where to submit issues though.
Note however that accessing tracks by index still works if project is not current (`project.tracks[index]`), even though it is less easy to use.

**Comment by Levitanus** [comment]
Now it's duplicate of #80, which is fixed with #77
